Reka llamaindex integration (llm and multi-modal-llm)

findalexli opened the pull request:
# Description

We are brining Reka to the llamaindex ecosystem via the `llama-index-llms-reka` and `llama-index-multi-modal-llms-reka` integration. Please refer to the readme [llama-index-llms-reka](https://github.com/reka-ai/llama_index/blob/6bc6dba1f38b9b96a90f4022ca8e8cb2cfa88794/llama-index-integrations/llms/llama-index-llms-reka/README.md) and [llama-index-llms-reka](https://github.com/reka-ai/llama_index/tree/main/llama-index-integrations/multi_modal_llms/llama-index-multi-modal-llms-reka) in each package for more documentation. 

We provided the integration tests and mocked-up tests that do not require Reka API remote access. 


## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ -  ] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [-  ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [-] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [-] I have performed a self-review of my own code
- [-] I have commented my code, particularly in hard-to-understand areas
- [-] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [- ] My changes generate no new warnings
- [-] I have added tests that prove my fix is effective or that my feature works
- [- ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

findalexli commented:
Hi there, 

1. let us now if you like us to take out the poetry lock file out of the commit
2. We have not yet run the `poetry build` and `poetry publish' to publish our file. let us know at which point would those command be executed - or would your team like to publish them instead.

logan-markewich commented on llama-index-integrations/llms/llama-index-llms-reka/llama_index/llms/reka/base.py:
```diff
@@ -0,0 +1,521 @@
+            )
+
+        self._client = Reka(api_key=api_key)
+        self._aclient = AsyncReka(api_key=api_key)
```
This will cause errors actually, let's move any `self.` to after the super init

logan-markewich commented on llama-index-integrations/llms/llama-index-llms-reka/poetry.lock:
we don't need to commit this, lets remove it (it just ends up taking up a lot of space)

logan-markewich commented on llama-index-integrations/llms/llama-index-llms-reka/pyproject.toml:
```diff
@@ -0,0 +1,57 @@
+# Feel free to un-skip examples, and experimental, you will just need to
+# work through many typos (--write-changes and --interactive will help)
+skip = "*.csv,*.html,*.json,*.jsonl,*.pdf,*.txt,*.ipynb"
+
```
lets add the tool.llamahub and tool.llamahub.class_authors sections (see other integrations for what this looks like)

logan-markewich commented on llama-index-integrations/llms/llama-index-llms-reka/pyproject.toml:
```diff
@@ -0,0 +1,57 @@
+llama-index-core = "^0.10.0"
+reka-api = "^3.0.8"
+pytest = "^7.2.1"
+pytest-asyncio = "^0.21.0"
```
shouldn't pytest and pytest-asyncio be in the dev dependencies?

logan-markewich commented on llama-index-integrations/multi_modal_llms/llama-index-multi-modal-llms-reka/poetry.lock:
lets delete this file

logan-markewich commented on llama-index-integrations/multi_modal_llms/llama-index-multi-modal-llms-reka/llama_index/multi_modal_llms/reka/base.py:
```diff
@@ -0,0 +1,447 @@
+            )
+
+        self._client = Reka(api_key=api_key)
+        self._aclient = AsyncReka(api_key=api_key)
```
as with the normal LLM, lets move this to after the init to avoid errors

findalexli replied:
addressed.

logan-markewich commented on llama-index-integrations/multi_modal_llms/llama-index-multi-modal-llms-reka/pyproject.toml:
```diff
@@ -0,0 +1,51 @@
+
+[tool.poetry.dependencies]
+python = ">=3.8.1,<4.0"
+llama-index-core = "^0.10.0"
```
actually, lets make the dep ^0.11.0 to match the latest release

findalexli replied:
Addressed.

logan-markewich commented on llama-index-integrations/multi_modal_llms/llama-index-multi-modal-llms-reka/pyproject.toml:
```diff
@@ -0,0 +1,51 @@
+python = ">=3.8.1,<4.0"
+llama-index-core = "^0.10.0"
+reka-api = "^3.0.8"
+pytest = "^8.3.2"
```
pytest should just be a dev dependency no?

logan-markewich commented on llama-index-integrations/multi_modal_llms/llama-index-multi-modal-llms-reka/pyproject.toml:
```diff
@@ -0,0 +1,51 @@
+# Feel free to un-skip examples, and experimental, you will just need to
+# work through many typos (--write-changes and --interactive will help)
+skip = "*.csv,*.html,*.json,*.jsonl,*.pdf,*.txt,*.ipynb"
+
```
lets add the tool.llamahub and tool.llamahub.class_authors sections here as well

logan-markewich reviewed (commented):
Looks good, just a few minor nits

findalexli commented on llama-index-integrations/llms/llama-index-llms-reka/pyproject.toml:
```diff
@@ -0,0 +1,57 @@
+
+[tool.poetry.dependencies]
+python = ">=3.8.1,<4.0"
+llama-index-core = "^0.10.0"
```
addressed

logan-markewich commented:
It looks like I can't push to this PR

You'll want to [install pants](https://www.pantsbuild.org/stable/docs/getting-started/installing-pants), run `pants tailor llama-index-integrations/::` and add the files it creates in your new packages

findalexli commented:
hi @logan-markewich  thanks! I added the build files generated by pants as you suggested. I also added you to our fork in case you need to push any changes.

findalexli commented:
hi @logan-markewich  we noticed that the unit test are not passing due to our api package `reka-api` not installed. however, we had specified them in the pyproject.toml file. What could be causing the issue?

findalexli commented:
we updated the build depency and run `pants test :: `without issue.  Can you please check
